src/code/z_view.c

```diff
@@ -279,137 +279,231 @@ void View_RenderView(View* view, s32 uParm2) {
-    guLookAt(view->viewingPtr, view->eye.x, view->eye.y, view->eye.z,
-             view->focalPoint.x, view->focalPoint.y, view->focalPoint.z,
+    OPEN_DISPS(gfxCtx);
+
+    vp = GRAPH_ALLOC(gfxCtx, sizeof(*vp));
+    View_ViewportToVp(vp, &view->viewport);
+    view->vp = *vp;
+
+    gDPPipeSync(OVERLAY_DISP++);
+    {
+        s32 pad;
+        Gfx* overlay;
+
+        overlay = OVERLAY_DISP;
+        View_WriteScissor(&overlay, view->viewport.leftX, view->viewport.topY, view->viewport.rightX,
+                          view->viewport.bottomY);
+        OVERLAY_DISP = overlay;
+    }
+
+    gSPViewport(OVERLAY_DISP++, vp);
+    projection = GRAPH_ALLOC(gfxCtx, sizeof(*projection));
+    view->projectionPtr = projection;
+
+    width = view->viewport.rightX - view->viewport.leftX;
+    height = view->viewport.bottomY - view->viewport.topY;
+    aspect = (f32)width / (f32)height;
+
+    guPerspective(projection, &view->normal, view->fovy, aspect, view->zNear, view->zFar, view->scale);
+    view->projection = *projection;
+
+    gSPPerspNormalize(OVERLAY_DISP++, view->normal);
+    gSPMatrix(OVERLAY_DISP++, projection, G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_PROJECTION);
+
+    viewing = GRAPH_ALLOC(gfxCtx, sizeof(*viewing));
+    view->viewingPtr = viewing;
+
+    if (view->eye.x == view->focalPoint.x && view->eye.y == view->focalPoint.y && view->eye.z == view->focalPoint.z) {
+        view->eye.z += 2.0f;
+    }
+
+    guLookAt(viewing, view->eye.x, view->eye.y, view->eye.z, view->focalPoint.x, view->focalPoint.y, view->focalPoint.z,
              view->upDir.x, view->upDir.y, view->upDir.z);
 
+    view->viewing = *viewing;
+
+    gSPMatrix(OVERLAY_DISP++, viewing, G_MTX_NOPUSH | G_MTX_MUL | G_MTX_PROJECTION);
+
+    CLOSE_DISPS(gfxCtx);
+
+    return 1;
+}
+
+s32 func_80140024(View* view) {
+    guLookAt(view->viewingPtr, view->eye.x, view->eye.y, view->eye.z, view->focalPoint.x, view->focalPoint.y,
+             view->focalPoint.z, view->upDir.x, view->upDir.y, view->upDir.z);
+
     view->unkE0 = *view->viewingPtr;
     view->viewingPtr = &view->unkE0;
 
     return 1;
 }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/code/z_view/func_801400CC.asm")
+s32 func_801400CC(View* view, Gfx** gfxp) {
+    Gfx* gfx = *gfxp;
+    GraphicsContext* gfxCtx = view->gfxCtx;
+    Viewport* viewport = &view->viewport;
+    Mtx* projection;
+    Vp* vp;
+
+    vp = GRAPH_ALLOC(gfxCtx, sizeof(*vp));
+    View_ViewportToVp(vp, viewport);
+    view->vp = *vp;
+
+    View_SyncAndWriteScissor(view, &gfx);
+
+    gSPViewport(gfx++, vp);
+
+    projection = GRAPH_ALLOC(gfxCtx, sizeof(*projection));
+    view->projectionPtr = projection;
+
+    guOrtho(projection, gScreenWidth * -0.5f, gScreenWidth * 0.5f, gScreenHeight * -0.5f, gScreenHeight * 0.5f,
+            view->zNear, view->zFar, view->scale);
+
+    view->projection = *projection;
+
+    gSPMatrix(gfx++, projection, G_MTX_LOAD | G_MTX_PROJECTION);
+    *gfxp = gfx;
 
+    return 1;
+}
```

Comment:
This was the trickiest function to get matching, I'm not super happy with it. 
I couldn't get it to match using the `OPEN_DISPS`/`CLOSE_DISPS` macros -- but maybe that's expected? Other functions which take a `Gfx**` also do not use these macros: probably the *caller* uses them?